src/server/injected/textSelectorEngine.ts

```diff
@@ -118,7 +118,7 @@ function queryInternal(root: SelectorRoot, matcher: Matcher, shadow: boolean): E
     if (!node)
       break;
     if (node.nodeType === Node.TEXT_NODE) {
-      lastText += node.nodeValue;
+      lastText += node.nodeValue ? node.nodeValue.replace(/\s/g, ' ') : node.nodeValue;
```

Comment:
Let's instead update `matcher` to cover `element.value` check below, and also similar checks in queryAllInternal.